compiler/rustc_mir_transform/src/check_unnecessary_transmutes.rs

```diff
@@ -89,7 +89,7 @@ impl<'tcx> Visitor<'tcx> for UnnecessaryTransmuteChecker<'_, 'tcx> {
             && let span = self.body.source_info(location).span
             && let Some(lint) = self.is_unnecessary_transmute(
                 func,
-                self.tcx.sess.source_map().span_to_snippet(arg).expect("ok"),
+                self.tcx.sess.source_map().span_to_snippet(arg).as_deref().unwrap_or("…"),
                 span,
             )
```

Comment:
Question: also, this fallback seems not ideal to me. I feel like we should not be making a *suggestion* if we don't have the source snippet to work with?

Reply:
im not sure. when i [searched up uses of span_to_snippet](https://github.com/search?q=repo%3Arust-lang%2Frust%20source_map().span_to_snippet&type=code), i found some examples of `let Ok() = ..to_snippet()` and some examples of `unwrap_or`... i think in the specific case of our repro, we do want this to lint, because it is part of the users problem.

Reply:
Or rather, stylistically, I would prefer to be faithful in `is_unnecessary_transmute` and report that we don't have a source snippet for `arg` to work with. And in `is_unnecessary_transmute`, there we *explicitly* change the suggestion level:

- If `arg` source snippet is `Some`, we make an `MaybeIncorrect` suggestion.
- However, if `arg` source snippet is N/A, we make an `HasPlaceholder` suggestion.

Reply:
EDIT: I don't mean not firing the *lint*, I mean the *suggestion* of the lint.

Reply:
i dont actually know if it matters, it seems that in our case it doesnt end up linting?

Reply:
Hm, I think in this ICE we never even reach `is_unnecessary_transmute`, because it's the callsite `span_to_snippet` that ICEs. Still, I don't think we should use a dummy value here to paper over not having an `arg` snippet to work with. Instead of doing the `MaybeIncorrect`/`HasPlaceholder` dance, I'm also okay with just skipping the suggestion if we don't have an `arg` snippet to work with.